uni/lib/view/home/widgets/exam_card.dart

```diff
@@ -1,163 +1,127 @@
-                type: exam.type,
-                reverseOrder: true,
-              ),
-            ],
-          ),
-        ),
-      ),
-    );
-  }
-}
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/model/entities/exam.dart';
+import 'package:uni/model/providers/lazy/exam_provider.dart';
+import 'package:uni/utils/drawer_items.dart';
+import 'package:uni/view/common_widgets/generic_card.dart';
+import 'package:uni/view/common_widgets/request_dependent_widget_builder.dart';
+import 'package:uni/view/home/widgets/exam_card_shimmer.dart';
+import 'package:uni/view/home/widgets/next_exams_card.dart';
+import 'package:uni/view/home/widgets/remaining_exams_card.dart';
+import 'package:uni/view/lazy_consumer.dart';
+
+class ExamCard extends GenericCard {
+  ExamCard({super.key});
+
+  const ExamCard.fromEditingInformation(
+    super.key, {
+    required super.editingMode,
+    super.onDelete,
+  }) : super.fromEditingInformation();
+
+  @override
+  String getTitle(BuildContext context) =>
+      S.of(context).nav_title(DrawerItem.navExams.title);
+
+  @override
+  Future<Object?> onClick(BuildContext context) =>
+      Navigator.pushNamed(context, '/${DrawerItem.navExams.title}');
+
+  @override
+  void onRefresh(BuildContext context) {
+    Provider.of<ExamProvider>(context, listen: false).forceRefresh(context);
+  }
+
+  @override
+  Widget buildCardContent(BuildContext context) {
+    return LazyConsumer<ExamProvider>(
+      builder: (context, examProvider) {
+        final filteredExams = examProvider.getFilteredExams();
+        final hiddenExams = examProvider.hiddenExams;
+        final allExams = filteredExams
+            .where((exam) => !hiddenExams.contains(exam.id))
+            .toList();
+        return RequestDependentWidgetBuilder(
+          status: examProvider.status,
+          builder: () => generateExams(allExams, context),
+          hasContentPredicate: allExams.isNotEmpty,
+          onNullContent: Center(
+            child: Text(
+              S.of(context).no_selected_exams,
+              style: Theme.of(context).textTheme.titleLarge,
+            ),
+          ),
+          contentLoadingWidget: const ExamCardShimmer().build(context),
+        );
+      },
+    );
+  }
+
+  Widget generateExams(List<Exam> allExams, BuildContext context) {
+    final nextExams = getPrimaryExams(
+      allExams,
+      allExams.isNotEmpty ? allExams.first : null,
+    );
+    final primaryExams = NextExamsWidget(exams: nextExams);
+
+    final remainingExamsCount = 4 - nextExams.length;
+    final List<Exam> remainingExams;
+    if (remainingExamsCount > 0) {
+      remainingExams = allExams
+          .where((exam) => !nextExams.contains(exam))
+          .take(remainingExamsCount)
+          .toList();
+    } else {
+      remainingExams = [];
+    }
+    final secondaryExams = RemainingExamsWidget(exams: remainingExams);

```

Comment:
This `4` is puzzling me. Is it the max total exams shown? Why not just filter the next exams and take the first 2/3 directly below.

Reply:
Regarding the use of the number "4" in the code, I introduced it as a variable for flexibility and easier maintenance. Given the dynamic nature of our dataset, the number of exams can vary significantly. I think implementing a fixed percentage could result in an exam display that is either too small or too large (?).